plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergFileOperations.java

```diff
@@ -509,6 +510,59 @@ public void testSelectWithFilter()
                         .build());
     }
 
+    @Test
+    public void testPartialPartitionPruningEffectivenessWithPartitionTransform()
+    {
+        testPartialPartitionPruningEffectivenessWithPartitionTransform("hour(d)", 4);
+        testPartialPartitionPruningEffectivenessWithPartitionTransform("day(d)", 2);
+        testPartialPartitionPruningEffectivenessWithPartitionTransform("month(d)", 2);
+        testPartialPartitionPruningEffectivenessWithPartitionTransform("year(d)", 2);
+        testPartialPartitionPruningEffectivenessWithPartitionTransform("bucket(d, 4)", 2);
+    }
+
+    private void testPartialPartitionPruningEffectivenessWithPartitionTransform(String partitionTransform, int expectedDataFileOperations)
+    {
+        String tableName = "test_day_transform_timestamp" + randomNameSuffix();
+        assertUpdate(format("CREATE TABLE %s (d TIMESTAMP(6), b BIGINT) WITH (partitioning = ARRAY['%s'])", tableName, partitionTransform));
+
+        @Language("SQL") String values =
+                """
+                VALUES
+                    (NULL, 101),
+                    (TIMESTAMP '1969-12-25 15:13:12.876543', 8),
+                    (TIMESTAMP '1969-12-30 18:47:33.345678', 9),
+                    (TIMESTAMP '1969-12-31 00:00:00.000000', 10),
+                    (TIMESTAMP '1969-12-31 05:06:07.234567', 11),
+                    (TIMESTAMP '1970-01-01 12:03:08.456789', 12),
+                    (TIMESTAMP '2015-01-01 10:01:23.123456', 1),
+                    (TIMESTAMP '2015-01-01 11:10:02.987654', 2),
+                    (TIMESTAMP '2015-01-01 12:55:00.456789', 3),
+                    (TIMESTAMP '2015-05-15 13:05:01.234567', 4),
+                    (TIMESTAMP '2015-05-15 14:21:02.345678', 5),
+                    (TIMESTAMP '2020-02-21 15:11:11.876543', 6),
+                    (TIMESTAMP '2020-02-21 16:12:12.654321', 7)
+                """;
+        assertUpdate("INSERT INTO " + tableName + " " + values, 13);
+        assertQuery("SELECT * FROM " + tableName, values);
+
+        @Language("SQL") String selectQuery = "SELECT * FROM " + tableName + " WHERE d >= TIMESTAMP '2015-05-15 01:23:45.678901'";
+        assertThat(query(selectQuery)).isNotFullyPushedDown(FilterNode.class);
+
```

Comment:
nit: We could add another assertion that ensures the number of generated files is greater than the `expectedDataFileOperations` likes:

```java
        assertThat((long) computeScalar("SELECT COUNT(DISTINCT file_path) FROM \"" + tableName + "$files\""))
                .isGreaterThan(expectedDataFileOperations);
```